bin/checkver.ps1

```diff
@@ -23,213 +43,216 @@ if (!$app -and $update) {
         continue
     }
 
-    if(!$regex -and $replace) {
-        write-host -f darkred "'replace' requires 're'"
+    if (!$regex -and $replace) {
+        next "'replace' requires 're' or 'regex'"
         continue
     }
 
-    if($jsonpath) {
+    if ($jsonpath) {
         $ver = json_path $page $jsonpath
-        if(!$ver) {
+        if (!$ver) {
             $ver = json_path_legacy $page $jsonpath
         }
-        if(!$ver) {
-            write-host -f darkred "couldn't find '$jsonpath' in $url"
+        if (!$ver) {
+            next "couldn't find '$jsonpath' in $url"
             continue
         }
     }
 
-    if($jsonpath -and $regexp) {
+    if ($jsonpath -and $regexp) {
         $page = $ver
-        $ver = ""
+        $ver = ''
     }
 
-    if($regexp) {
-        $regex = new-object System.Text.RegularExpressions.Regex($regexp)
-        if($reverse) {
-            $match = $regex.matches($page) | select-object -last 1
+    if ($regexp) {
+        $regex = New-Object System.Text.RegularExpressions.Regex($regexp)
+        if ($reverse) {
+            $match = $regex.Matches($page) | Select-Object -Last 1
         } else {
-            $match = $regex.matches($page) | select-object -first 1
+            $match = $regex.Matches($page) | Select-Object -First 1
         }
 
-        if($match -and $match.Success) {
+        if ($match -and $match.Success) {
             $matchesHashtable = @{}
             $regex.GetGroupNames() | ForEach-Object { $matchesHashtable.Add($_, $match.Groups[$_].Value) }
             $ver = $matchesHashtable['1']
             if ($replace) {
-                $ver = $regex.replace($match.Value, $replace)
+                $ver = $regex.Replace($match.Value, $replace)
             }
-            if(!$ver) {
+            if (!$ver) {
                 $ver = $matchesHashtable['version']
             }
         } else {
-            write-host -f darkred "couldn't match '$regexp' in $url"
+            next "couldn't match '$regexp' in $url"
             continue
         }
     }
 
-    if(!$ver) {
-        write-host -f darkred "couldn't find new version in $url"
+    if (!$ver) {
+        next "couldn't find new version in $url"
         continue
     }
 
-    if($ver -eq $expected_ver -and $forceUpdate -eq $false) {
-        # version hasn't changed (step over if forced update)
-        write-host "$ver" -f darkgreen
+    # Skip actual only if versions are same and there is no -f
+    if (($ver -eq $expected_ver) -and !$ForceUpdate -and $SkipSupported) { continue }
+
+    Write-Host "$App`: " -NoNewline
+
+    # version hasn't changed (step over if forced update)
+    if ($ver -eq $expected_ver -and !$ForceUpdate) {
+        Write-Host $ver -ForegroundColor DarkGreen
         continue
     }
 
-    write-host "$ver" -f darkred -nonewline
-    write-host " (scoop version is $expected_ver)" -NoNewline
-    $update_available = (compare_versions $expected_ver $ver) -eq -1
```

Comment:
`$updateAvailable` or `$UpdateAvailable` or `$update_available`. Combining snake_case and camelCase looks awkward 😁

Reply:
Wrong regex for rename all occurancies

Reply:
Why is there this `$update_available` check? You already know, there is version mismatch if you use `$ver -eq $expected_ver`

Reply:
It is only to determine if it's upgrade or downgrade? 

![c](https://i.imgur.com/LolxUYm.png)